base/setup/welcome/welcome.c

```diff
@@ -275,6 +370,10 @@ PTOPIC AddNewTopicEx(
     {
         pTopic->bIsCommand = TRUE;
         StringCchCopy(pTopic->szCommand, ARRAYSIZE(pTopic->szCommand), szCommand);
+
+        /* Check for special applications: ReactOS Installer */
+        if (_tcsicmp(pTopic->szCommand, TEXT("reactos.exe")) == 0)
+            ExpandInstallerPath(pTopic->szCommand, ARRAYSIZE(pTopic->szCommand));
```

Comment:
Why does this have a special case?

Reply:
Because it's the only app that is specified, in the welcome.ini / resource file, with no complete path, and because it's the installer I want to be able to find it in the "correct" platform-specific subdirectory (if there is one) on the CD-ROM (eg. \i386).

Reply:
In that case, wouldnt it make sense to look in the platform specific subdir first?

Reply:
You're right, it would make much more sense. And then I "fall back" to the generic case if I cannot start the installer from the platform-specific subdir.